### PR TITLE
Make players ignore no-pvp fields damage

### PR DESCRIPTION
**Pull request by marmichalski**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Always ignore no-pvp fields damage.

**Issues addressed:** #4269

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/combat.cpp [hunk 1]
```diff
@@ -1461,7 +1474,7 @@ void MagicField::onStepInField(Creature* creature)
 			}
 
 			Player* targetPlayer = creature->getPlayer();
-			if (targetPlayer) {
+			if (!harmfulField && targetPlayer) {
```
[the remark this answers is not in the document]
Comment (Infolinia) [reply src/combat.cpp:1477]: Hello, non pvp title works good. But, now pvp field in remere map editor nie zabiera zdrowia. Shows CONST_ME_POFF in stay in pvp title.

**Comment by Infolinia** [comment]
Hello, non pvp title works good. But, now pvp field in remere map editor nie zabiera zdrowia. Shows CONST_ME_POFF in stay in pvp title.

**Comment by Infolinia** [comment]
Hello, non pvp title works good. But, now pvp field in remere map editor does not take away health. Shows CONST_ME_POFF in stay in pvp title.

**Comment by marmichalski** [comment]
> Hello, non pvp title works good. But, now pvp field in remere map editor does not take away health. Shows CONST_ME_POFF in stay in pvp title.

🤷 
pushed fix, can you check again?

**Comment by Infolinia** [comment]
Don't work. :( Look this film: [https://youtu.be/EhF_ghUJxlA](url)

**Comment by marmichalski** [comment]
> Don't work. :( Look this film: [https://youtu.be/EhF_ghUJxlA](url)

🙈 should be almost fantastic now

**Comment by Infolinia** [comment]
Great, works fine! Thank you for help.

**Review by EPuncker — APPROVED** [review]

**Review by ranisalt — APPROVED** [review]
LGTM

**Pull request closed without merging**